pkg/devfile/parser/data/v2/commands.go

```diff
@@ -57,3 +53,24 @@ func (d *DevfileV2) UpdateCommand(command v1.Command) {
 		}
 	}
 }
+
+// DeleteCommand removes the specified command
+func (d *DevfileV2) DeleteCommand(id string) error {
+
+	found := false
+	for i := len(d.Commands) - 1; i >= 0; i-- {
```

Comment:
can use `for index := range d.Commands` to find the index
and `append(d.Commands[:index], slice[index+1:]...)`